### DISCUSSION TITLE
Object book talk

### DISCUSSION THREAD

**Discussion by jongfeel** [ThinkAboutSoftware/OnlineSelfCodingGroup#22] [Book talk]
## Overview

@yoonucho 님이 마침 제가 읽는 책과 같은 책을 읽고 있고 있어서 알게 되었고 #20 

제가 좋아하는 주제이기도 해서 대화의 시간을 마련해 보고자 discussions 탭에 만들게 되었습니다.

논의할 범위와 주제를 선정한 후에, 언제 얘기할지 오프라인이 좋을지 온라인이 좋을지도 정해보면 좋겠습니다.

**Comment by yoonucho** [comment]
@jongfeel  멘토님 저는 우선 온라인이 좋을거 같고 시간은 평일이라면 7시 이후면 괜찮을거 같습니다.

범위와 주제는 저는 66페이지까지 봤고 멘토님도 읽으신 페이지를 알려주시면  그 안에서  정했으면 좋겠습니다!

**Reply by jongfeel** [reply]
저는 책 읽은 건 chapter 03 까지 읽었습니다. 읽다가 C# 코드로 바꾸면 좋겠다는 생각이 너무 강하게 들어서 조영호님에게 허락을 구하고 따로 정리 작업을 하고 있었거든요.

그러면 `chapter 02 객체지향 프로그래밍` 까지를 범위로 잡고 논의할 주제를 자유롭게 적고 얘기를 해보면 좋겠습니다.

주제를 정리하고 얘기할 거리를 만들 문서를 준비해야 하니
다음 주 화요일(2월 9일)이 어떠신가요? 시간은 물론 오후 11시 입니다.

준비한 분량이 맘에 안들거나 다른 일정이 생겨서 참석을 못하거나 하면 언제든 일정 조절은 가능합니다.

문서는 이 repository에 md 파일로 넣어 두면 좋을 것 같습니다.
pull request를 줘서 서로 미리 검토할 수 있게 하면 더 좋을 것 같아요.

**Reply by yoonucho** [reply]
네 좋습니다. 기간내에 열심히 준비해보겠습니다!

**Comment by jongfeel** [comment]
#24 를 통해 제 논의 내용을 올렸습니다.

**Comment by yoonucho** [comment]
@jongfeel  멘토님 정말 죄송한데 일정을 설날 연휴 기간이나 이후로 변경할 수 없을까요 연휴 전에 완료해야 할 업무가 있어서요 ㅜㅜ

**Reply by jongfeel** [reply]
야근이 많으신가 보군요. 그러면 다음 주로 일정을 미뤄 두겠습니다.

미리 논의 내용을 pull request로 주시고 서로 리뷰를 하는 과정을 거쳐야 원활한 대화가 가능할 것 같습니다.

**Reply by yoonucho** [reply]
네  미팅시간전에  내용은 마저 정리해서  PR드리겠습니다.  감사합니다.  ㅠㅠ

**Comment by yoonucho** [comment]
#25 에 제가 생각하고 기록한 내용을 올렸습니다.

**Comment by jongfeel** [comment]
어제 오늘 진행중인 pull request #25 review conversation은 윤우님의 논의에 대한 review 입니다.

이후 추가로 해야 할 내용과 규칙을 얘기해 드리면
- 저의 pull request #24 에도 review가 진행되어야 합니다.
- 하나의 review comment는 `Conversation`으로 대체됩니다. 그래서 논의가 충분히 이루어졌다고 생각되는 사람이 `Resolve conversation`을 하시면 됩니다. 제가 논의가 이루어졌다고 생각하는 conversation은 모두 resolve를 걸어 뒀습니다.
- 
![image](https://user-images.githubusercontent.com/17442457/107909320-4600be80-6f9b-11eb-9f4c-98b433e10cb2.png)

- 만약 논의가 더 필요하다고 판단되면 `Show conversation` 버튼을 누르고 `Unresolved conversation` 버튼을 누르면 계속해서 대화가 진행 가능합니다.
- 
![image](https://user-images.githubusercontent.com/17442457/107909482-a42da180-6f9b-11eb-95d5-714bef5bf514.png)
- 
![image](https://user-images.githubusercontent.com/17442457/107909502-b90a3500-6f9b-11eb-9f7d-9d404cb28f67.png)

**Reply by yoonucho** [reply]
멘토님 저도 2시간전에 멘토님 의견에 리뷰를 달긴 했는데 저는 멘토님글에 Resolve conversation버튼이 보이지 않습니다.
두명이상이 대화를 주고 받은 이후에 생기는  버튼인가요?

**Reply by jongfeel** [reply]
있는데요??? 으흠

**Reply by yoonucho** [reply]
첫번째 스샷은  멘토님의 글이고  conversation 버튼이 안보입니다.

![화면 캡처 2021-02-15 211208](https://user-images.githubusercontent.com/2981954/107945471-048c0580-6fd3-11eb-8190-765dfcb2b56e.jpg)

두번째 스샷은 제글인데 conversation이 보입니다.
![화면 캡처 2021-02-15 211547](https://user-images.githubusercontent.com/2981954/107945477-05bd3280-6fd3-11eb-97c7-2dfed354c653.jpg)

권한문제일까요?

**Reply by jongfeel** [reply]
권한문제 맞는 것 같습니다

**Reply by jongfeel** [reply]
방금 write 권한으로 바꿨어요

**Comment by jongfeel** [comment]
### 다음 book talk 계획

- Talk 날짜: 3월 9일 화요일
- Pull request 완료 날짜: 3월 5일 금요일 까지
- chapter 03, chapter 04
- 윤우님의 경우 prologue와 chapter 01 추가
- chapter 03과 chapter 04를 각각 하나의 pull request로 해서 리뷰 진행
- 원활한 conversation을 위해, talk 3일전 까지 pull request가 완료 되어야 함.

**Comment by jongfeel** [comment]
첫 번째 book talk review에 아직 conversation이 열려 있는 항목이 몇 개 있습니다.
https://github.com/ThinkAboutSoftware/OnlineSelfCodingGroup/pull/25

시간 되실 때 (적어도 다음 book talk 전까지) 추가 의견을 적어 주시고,
resolve가 가능한 부분에 대해서는 resolve 해주셔도 됩니다.

**Reply by yoonucho** [reply]
네 알겠습니다!

**Comment by yoonucho** [comment]
멘토님 죄송한데 book talk 날짜 변경 요청드립니다. 다음 주 중에 멘토님이 스케쥴 없는 날로 변경부탁드릴게요

**Reply by jongfeel** [reply]
알겠습니다. 1주 미뤄서 다음 주 화요일로 변경할께요.

**Reply by yoonucho** [reply]
감사합니다  😭

**Comment by yoonucho** [comment]
멘토님 정말 죄송한데요 염치없지만  book talk 날짜 변경 부탁드릴게요 왜냐면 좀전에 엄마가 실수로 모뎀선을  잘라서요 ㅋㅋㅜㅠ ㅠㅠ

**Reply by jongfeel** [reply]
으악 어쩌다 이런 일이!

선 새로 사서 하려면 최소 하루나 이틀은 필요하겠군요.

새로 선 세팅 되면 알려주세요. 이번 주 내에 할 수 있다면 전 가능합니다.

**Comment by yoonucho** [comment]
네 인터넷기사님이 내일 오전중으로 방문하신다 해서 내일도 가능할거 같습니다 자세한건 내일 처리후에 말씀드릴게요 ㅋ

**Reply by jongfeel** [reply]
네 그럼 일단 내일로 옮겨 놓겠습니다.

**Reply by yoonucho** [reply]
멘토님 모뎀선 복구되었습니다. 오늘 북토크 완전 가능합니다!  <😹>
